clients/lsp-csharp.el

```diff
@@ -108,22 +116,16 @@ Usually this is to be set in your .dir-locals.el on the project root directory."
 (lsp-dependency
  'omnisharp-roslyn
  `(:download :url lsp-csharp-omnisharp-roslyn-download-url
-   :store-path lsp-csharp-omnisharp-roslyn-store-path)
+             :decompress :zip
+             :store-path lsp-csharp-omnisharp-roslyn-store-path
+             :binary-path lsp-csharp-omnisharp-roslyn-binary-path
+             :set-executable? t)
  '(:system "OmniSharp"))
 
 (defun lsp-csharp--omnisharp-download-server (_client callback error-callback _update?)
   "Download zip package for omnisharp-roslyn and install it.
 Will invoke CALLBACK on success, ERROR-CALLBACK on error."
-  (lsp-package-ensure
-   'omnisharp-roslyn
-   (lambda ()
-     (lsp-unzip lsp-csharp-omnisharp-roslyn-store-path
-                lsp-csharp-omnisharp-roslyn-server-dir)
-     (unless (eq system-type 'windows-nt)
-       (let ((omnisharp-executable (f-join lsp-csharp-omnisharp-roslyn-server-dir "OmniSharp")))
-         (set-file-modes omnisharp-executable #o755)))
-     (funcall callback))
-   error-callback))
+  (lsp-package-ensure 'omnisharp-roslyn callback error-callback))
```

Comment:
Does the `:system "OmniSharp"` will Use `OmniSharp.exe` for windows?

Reply:
https://github.com/emacs-lsp/lsp-mode/pull/3857/files#diff-d1c03bcd1ccd0772a57b998e7b7278be607a4edefc38c27209c371d8a2504bffR123

Reply:
Can someone on windows confirm if `M-: (executable-find "bynary-name")` works without `.exe`?